src/Mod/TechDraw/Gui/QGIViewBalloon.cpp

```diff
@@ -765,6 +765,11 @@ void QGIViewBalloon::draw()
     dLinePath.lineTo(arrowTipX - xAdj, arrowTipY - yAdj);
     balloonLines->setPath(dLinePath);
 
+    if (!balloon->LineVisible.getValue()) {
```

Comment:
Why just an `if`, and not an `if-else`? You could disable the line and end shape drawing entirely instead of just overwriting them here.

Reply:
Good question. I did it that way first, but the if-block ended up spanning a huge block of code, which kind of destroyed the code readibility.  At least on my screen, you couldn't see both ends of the if-block at the same time, which in my opinion makes the code difficult to read.  So, I decided to choose the option with better code readability over the (slightly) increased performance.  I realized that you'd have to have at least tens of thousand of these balloons with hidden lines on your TD page to notice a performance difference.

However, in principle you are right, but I'd suggest that someone would refactor the code to be both readable and optimal.  At this point, I wanted to avoid making any drastic change, I've just started making FreeCAD code contributions for fun, and I'm kind of trying to start it small.

Reply:
Yeah @chennes 's suggestion is an optimization. We can go step by step of course.
Would you mind just adding a comment in the code at that point, so anyone else reading it would get an idea of how this might be improved?

Reply:
> Would you mind just adding a comment in the code at that point, so anyone else reading it would get an idea of how this might be improved?

That's a fair idea.  I planned to "later" refactor the function, but I'd probably forgot about it anyways, so it's good to make a comment for future coders, as you said.  So, I simply added a comment.

Reply:
Thanks! I'm merging now. I like the idea that we're not forced to solve all of the earth's problems in one go, but we leave marks in the code for other people (or ourselves) to take on :) Another goo thing to do too is using "TODO" word in comments as some IDEs or CI systems look for that word to find issues in the code.